.ado/variables/msbuild.yml

```diff
@@ -1,6 +1,4 @@
 variables:
-  MSBuildArchitecture: x64
```

Comment:
> MSBuildArchitecture: x64 [](start = 1, length = 25)

I'm not familiar with the .yml include structure, but assuming the variables here can be included in all other .yml files: Wouldn't it make sense to keep these variables and ensure they get used everywhere (instead of literal platform variables)? This way, we'd have a central switch.

Reply:
We ideally shouldn't specialize it anywhere in CI logic. I.e. we should keep as much build logic as possible outside of CI configuration, and inside of build logic. 

Historically when we've asked CI to do configuration, it gets out of sync with what users run, and we find regressions that CI didn't catch.